src/components/AddressSearch/index.tsx

```diff
@@ -407,14 +450,39 @@ function AddressSearch(
                                     setDisplayListViewBorder(false);
                                 }
                             },
+                            onKeyPress: (e: KeyboardEvent) => {
+                                if (!e) {
+                                    return;
+                                }
+                                if (e.code === 'ArrowUp') {
```

Comment:
Why do you need to manage the arrow keys outside of `useArrowKeyFocusManager`?

Reply:
@roryabraham The arrowkeyfocusmanager can only focus on search result items but the current location item is not included in the search results.

Reply:
@jsdev2547 I too agree with the question above.

We already have,

```
const [focusedIndex, setFocusedIndex] = useArrowKeyFocusManager({
        initialFocusedIndex: -1,
        maxIndex: maxIndex - 1,
        isActive: true,
    });
```

`focusedIndex` here represent the index including the currentLocation header and the search list. 
Why can't we have it like, when the focusIndex == 0, focus currentLocation button, and the rest for the suggestion list. 
We can have the `maxIndex as suggestionCount + currentLocationItem` when we show currentLocation. (Throwing idea on high level.)

I am not sure of the need by we need to listen to 'ArrowUp' here also, which is what `useArrowKeyFocusManager` does and returns us the focusedIndex.

Reply:
@abdulrahuman5196 I tried to use onFocusedIndexChange callback of useArrowKeyFocusManager instead of the onKeyPress event but the arrowkeyfocusmanager has reserved index 0 for search list and the focusindex -1 is when the search list is empty, so even if maxIndex is set to 1 to cover for currentLocationItem, the focusIndex would reset to 0 when the search is entered and focusIndex 0 would be for the search list. The arrowkeyfocusmanager has some flexibility and can leverage onFocusedIndexChange but the difficulty is presented by the current location item which is not part of the search list.

Reply:
@roryabraham @abdulrahuman5196 I suggest to remove the ArrowKeyFocusManager and use KeyPress event instead. I have tested using KeyPress event and the ArrowKeyFocusManager is not needed. The ArrowKeyFocusManager only focuses if the search result is not empty but the current location item needs focus when the search list is empty.

Reply:
IMO ArrowKeyFocusManager is just a effective wrapper to handle the key press. It would be better to use it or update it if required.
@jsdev2547 Could you kindly provide the code pointers for the above concern/analysis? Let me also see if there is something else we can do?

Reply:
@abdulrahuman5196 Please get latest and review.

Reply:
@abdulrahuman5196 I have pushed the revised KeyPress event handler. Please get latest and review.

Reply:
@abdulrahuman5196 I have pushed the revised KeyPress event handler. Please review.